zookeeper-api/src/main/java/org/apache/helix/zookeeper/zkclient/callback/ZkAsyncCallbacks.java

```diff
@@ -259,10 +276,15 @@ public int getRc() {
       return _rc;
     }
 
+    public String getPath() {
+      return _path;
+    }
+
     @Override
     public void notifyCallers() {
       LOG.warn("The callback {} has been cancelled.", this);
       markOperationDone();
+      handle();
```

Comment:
This confuses me. If we already mark operations are done, then why we need to call handle. Shall we put handle before operation done mark?

Also in previous function logic, mark operation done is in finally statement. It is lastly called after handle(). This logic is changed order.

Reply:
Thanks for the suggestion. Putting handle() before markOperationDone() make sense.